opensrs/opensrsapi.py

```diff
@@ -405,6 +411,8 @@ def register_domain(self, domain, purchase_period, user, user_id,
                             'Invalid syntax on domain')):
                     raise errors.InvalidDomain(e)
                 raise errors.DomainRegistrationFailure(e)
+            if e.response_code == self.CODE_CLIENT_TIMED_OUT:
+                raise errors.DomainRegistrationTimedOut(e)
```

Comment:
I see... Just thinking.  What if we add specific errors only for codes we ever saw in DS? In other words, for real use cases.